backend/src/routes/dashboard.js

```diff
@@ -1,21 +1,22 @@
 const express = require('express');
 
+const validateUser = require('../middleware/validateUser');
 const DashboardController = require('../controllers/dashboard');
 
 const router = express.Router();
 
-router.get('/stock-overview/:id', DashboardController.getStockOverview);
+router.get('/stock-overview/:symbol', validateUser, DashboardController.getStockOverview);
 
-router.get('/time-series/intraday/:id', DashboardController.getTimeSeriesIntraday);
+router.get('/time-series/intraday/:symbol', validateUser, DashboardController.getTimeSeriesIntraday);
 
-router.get('/time-series/daily/:id', DashboardController.getTimeSeriesDaily);
+router.get('/time-series/daily/:symbol', validateUser, DashboardController.getTimeSeriesDaily);
 
-router.get('/time-series/weekly/:id', DashboardController.getTimeSeriesWeekly);
+router.get('/time-series/weekly/:symbol', validateUser, DashboardController.getTimeSeriesWeekly);
 
-router.get('/time-series/monthly/:id', DashboardController.getTimeSeriesMonthly);
+router.get('/time-series/monthly/:symbol', validateUser, DashboardController.getTimeSeriesMonthly);
 
-router.get('/trending-stocks', DashboardController.getTrending);
+router.get('/trending-stocks', validateUser, DashboardController.getTrending);
 
-router.get('/predict-price/:id', DashboardController.predictPrice);
+router.get('/predict-price/:symbol', validateUser, DashboardController.predictPrice);
```

Comment:
Does this validateUser get used anywhere within the methods?

Reply:
yes, take a look at the validateUser function in middleware